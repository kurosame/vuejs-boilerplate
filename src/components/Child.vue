<template>
  <div class="child">
    <div>
      <span data-test="count">{{ counter.count }}</span>
      <button data-test="add-count" @click="$emit('addCount')">ADD</button>
    </div>
    <div>
      <span data-test="axios-count">{{ counter.axiosCount }}</span>
      <button data-test="add-axios-count" @click="$emit('addAxiosCount')">ADD</button>
      ※axios sample
    </div>
    <div>
      <span data-test="async-await-count">{{ counter.asyncAwaitCount }}</span>
      <button data-test="add-async-await-count" @click="$emit('addAsyncAwaitCount')">ADD</button>
      ※async await sample
    </div>
  </div>
</template>
<style lang="scss" scoped>
.child {
  color: white;
  background-color: blue;
}
</style>
<script lang="ts">
/* tslint:disable:prettier */
// TODO: https://github.com/kurosame/vuejs-boilerplate/issues/1
/* tslint:enable:prettier */
import { ICounterState } from '@/vuex/state/counter'
import Vue from 'vue'

type Prop<T> = () => T

export default Vue.extend({
  name: 'Child',
  props: {
    counter: Object as Prop<ICounterState>
  }
})
</script>
