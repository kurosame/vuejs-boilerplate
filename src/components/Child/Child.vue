<template src="./child.html"></template>
<style scoped src="./child.css"></style>
<script src="./child.js"></script>
