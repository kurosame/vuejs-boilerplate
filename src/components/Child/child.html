<div class="child">
  <div>
    <span class="count">{{ count }}</span>
    <button type="button" class="btn btn-success" @click="addValue">ADD</button>
  </div>
  <div>
    <span class="axios-count">{{ axiosCount }}</span>
    <button type="button" class="btn btn-success" @click="axiosSample">ADD</button>
    ※axios sample
  </div>
  <div>
    <span class="async-await-count">{{ asyncAwaitCount }}</span>
    <button type="button" class="btn btn-success" @click="asyncAwaitSample">ADD</button>
    ※async await sample
  </div>
</div>