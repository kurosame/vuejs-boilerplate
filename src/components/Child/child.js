export default {
  name: 'Child',
  props: {
    addValue: {
      type: Function,
      required: true
    },
    count: {
      type: Number,
      required: true
    }
  }
}
