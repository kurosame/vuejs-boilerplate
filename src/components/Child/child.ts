import Vue from 'vue'

export default Vue.extend({
  name: 'Child',
  props: [
    'addValue',
    'axiosSample',
    'asyncAwaitSample',
    'count',
    'axiosCount',
    'asyncCount'
  ]
})
