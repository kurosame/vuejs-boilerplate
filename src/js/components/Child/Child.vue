<template src="./child.html"></template>
<style scoped lang="scss" src="./child.scss"></style>
<script src="./child.js"></script>
