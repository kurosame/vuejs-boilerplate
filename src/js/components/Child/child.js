export default {
  name: 'Child',
  props: {
    addValue: {
      type: Function,
      required: true
    },
    axiosSample: {
      type: Function,
      required: true
    },
    asyncAwaitSample: {
      type: Function,
      required: true
    },
    count: {
      type: Number,
      required: true
    },
    axiosCount: {
      type: Number,
      required: true
    },
    asyncCount: {
      type: Number,
      required: true
    }
  }
}
