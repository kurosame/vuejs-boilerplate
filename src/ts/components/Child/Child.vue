<template src="./child.html"></template>
<style scoped lang="scss" src="./child.scss"></style>
<script lang="ts" src="./child.ts"></script>
