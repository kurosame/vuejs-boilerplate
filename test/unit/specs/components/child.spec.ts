import Child from '@/components/Child.vue'
import { mount } from '@vue/test-utils'

const wrapper = mount(Child, {
  propsData: {
    count: 1,
    axiosCount: 2,
    asyncAwaitCount: 3
  }
})

test('Data binding from the propsData.count to the count', () => {
  expect(wrapper.html()).toContain('<span class="count">1</span>')
})

test('Click the button.add-value will emit the addValue', () => {
  expect(wrapper.emitted('addValue')).toBeUndefined()

  wrapper.find('button.add-value').trigger('click')

  expect(wrapper.emitted('addValue')).toBeTruthy()
  expect(wrapper.emitted('addValue')[0]).toEqual([])
})

test('Data binding from the propsData.axiosCount to the axiosCount', () => {
  expect(wrapper.html()).toContain('<span class="axios-count">2</span>')
})

test('Click the button.axios-sample will emit the axiosSample', () => {
  expect(wrapper.emitted('axiosSample')).toBeUndefined()

  wrapper.find('button.axios-sample').trigger('click')

  expect(wrapper.emitted('axiosSample')).toBeTruthy()
  expect(wrapper.emitted('axiosSample')[0]).toEqual([])
})

test('Data binding from the propsData.asyncAwaitCount to the asyncAwaitCount', () => {
  expect(wrapper.html()).toContain('<span class="async-await-count">3</span>')
})

test('Click the button.async-await-sample will emit the asyncAwaitSample', () => {
  expect(wrapper.emitted('asyncAwaitSample')).toBeUndefined()

  wrapper.find('button.async-await-sample').trigger('click')

  expect(wrapper.emitted('asyncAwaitSample')).toBeTruthy()
  expect(wrapper.emitted('asyncAwaitSample')[0]).toEqual([])
})

test('Match the snapshot', () => {
  expect(wrapper.html()).toMatchSnapshot()
})
