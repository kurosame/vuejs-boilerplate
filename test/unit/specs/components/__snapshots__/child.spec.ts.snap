// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot 1`] = `
<div class="child">
  <div><span class="count">1</span>
    <button type="button" class="btn btn-success add-value">ADD</button>
  </div>
  <div><span class="axios-count">2</span>
    <button type="button" class="btn btn-success axios-sample">ADD</button>
    ※axios sample
  </div>
  <div><span class="async-await-count">3</span>
    <button type="button" class="btn btn-success async-await-sample">ADD</button>
    ※async await sample
  </div>
</div>
`;
